frontends/web/src/components/auth/authrequired.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useTranslation } from 'react-i18next';
+import { View, ViewHeader } from '../view/view';
+import style from './authrequired.module.css';
+import { useEffect, useState } from 'react';
+import { authenticate, subscribeAuthErr, subscribeAuthOk, subscribeAuthRequired } from '../../api/backend';
+import { unsubscribe } from '../../utils/subscriptions';
+
+export const AuthRequired = () => {
+  const { t } = useTranslation();
+
+  const [authRequired, setAuthRequired] = useState(false);
+
+  const performAuth = () => {
+    setAuthRequired(true);
+    authenticate();
+  };
+
+  useEffect(() => {
+    performAuth();
+
+    const subscriptions = [
+      subscribeAuthRequired(performAuth),
+      subscribeAuthErr(performAuth),
+      subscribeAuthOk(() => setAuthRequired(false)),
+    ];
+    return () => unsubscribe(subscriptions);
+  }, []);
+
+  return (
+    <div className={[style.auth, authRequired ? style.active : style.hidden].join(' ')}>
```

Comment:
instead of `style.hidden` you could just return null?

`if (!autRequired) { return null; }`

to not put anything into the DOM.